packages/shopping-cart/test/cart-actions.tsx

```diff
@@ -129,19 +135,27 @@ async function setCart( cartKey: string, newCart: RequestCart ): Promise< Respon
 
 function ProductList( {
 	initialProducts,
+	initialProductsForReplace,
 	initialCoupon,
 }: {
-	initialProducts?: RequestCartProduct[];
+	initialProducts?: MinimalRequestCartProduct[];
+	initialProductsForReplace?: MinimalRequestCartProduct[];
 	initialCoupon?: string;
 } ) {
-	const { responseCart, addProductsToCart, applyCoupon } = useShoppingCart();
+	const { responseCart, addProductsToCart, replaceProductsInCart, applyCoupon } = useShoppingCart();
 	const hasAddedProduct = useRef( false );
 	useEffect( () => {
 		if ( initialProducts && ! hasAddedProduct.current ) {
 			hasAddedProduct.current = true;
 			addProductsToCart( initialProducts );
 		}
 	}, [ addProductsToCart, initialProducts ] );
+	useEffect( () => {
+		if ( initialProductsForReplace && ! hasAddedProduct.current ) {
```

Comment:
Should this have its own ref?

Reply:
Maybe but I don't have anywhere where we're doing both props at the same time right now.